message-transport/pkg/transport/transport.go

```diff
@@ -19,12 +19,15 @@ package transport
 
 import (
 	"github.com/projectriff/riff/message-transport/pkg/message"
+	"io"
 )
 
 //go:generate mockery -name=Producer -output mocktransport -outpkg mocktransport
 
 // Producer is an interface for sending messages to arbitrary topics.
 type Producer interface {
+	io.Closer
```

Comment:
Not sure I understand why this needs to be forced on clients. Why not dynamically test for it?

Reply:
It's part of the contract and, if a client wants to be well-behaved, it should close the producer when it has finished with it.

What do you see as being "forced" on clients? They are free _not_ to call `Close()`.

Reply:
Sorry, shouldn't have said "clients". Implementations rather: All Producers are now required to implement Close, even if it's a noop.

Instead, everywhere we (wanna) close, we could dynamically test that an impl is a Closer, or am I missing something?

Reply:
Agreed. I made the change. Thanks.